backend/internal/domain/services/authentication_service.go

```diff
@@ -0,0 +1,44 @@
+package services
+
+import (
+	"errors"
+	"fmt"
+	"shodo/internal/domain/helpers"
+	"shodo/internal/repository"
+	"shodo/models"
+
+	"golang.org/x/net/context"
+)
+
+type AuthenticationService struct {
+	Repository    repository.Users
+	TokensService *TokensService
+}
+
+func (this *AuthenticationService) LogIn(ctx context.Context, request models.LoginUserRequest) (*models.AuthTokens, error) {
```

Comment:
You should never use receiver as `this` or `self`
It is an abbreviation of the type it receives. So here, I'd use `s` here to signal the service.

Reply:
Fixed)